content/client-lib-development-guide/features.textile

```diff
@@ -367,6 +367,9 @@ h3(#realtime-connection). Connection
 ** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport. The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message. The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
 ** @(RTN19b)@ If there are any pending channels i.e. in the @ATTACHING@ or @DETACHING@ state, the respective @ATTACH@ or @DETACH@ message should be resent to Ably
 ** @(RTN19b)@ If a @SYNC@ is underway, ensure the client library adheres to @RTP3@
+* @(RTN20)@ Heartbeats
+** @(RTN20a)@ If a transport does not receive a heartbeat for a period greater than the sum of the @heartbeatPeriod@ (sent in the @connectionDetails@ in the @CONNECTED@ message received on that transport) and the @realtimeRequestTimeout@, that transport may be disconnected. This requirement is optional; in deciding whether to implement, client library developers should take into account whether the transport in question is susceptible to undetected dropped connections.
```

Comment:
> Why not state that once `realtimeRequestTimeout` has failed, the client initiates a Heartbeat and if a response is not received within `realtimeRequestTimeout` then it disconnects the connection.

OK. TBH I'm not convinced the effect of that would be much different from just having a timeout of 20s in the first place, unless a setTimeout callback in node under load can be delayed by 10s *more* than an external request, which I don't think is possible in an event loop. But shrug, it's harmless, so if you'd prefer that scheme, I'm happy to change it. Or go up to a 15/20s timeout. @paddybyers any opinion?

> BTW. Why did you not include an option to include an ID in Heartbeats so that clients can correlate the Heartbeat they sent with the one they received? 

Happy to add it. (Missed it out because these changes are about server-sent heartbeats only, I didn't touch the pings section.)